api/envoy/extensions/filters/udp/udp_proxy/session/dynamic_forward_proxy/v3/dynamic_forward_proxy.proto

```diff
@@ -0,0 +1,54 @@
+syntax = "proto3";
+
+package envoy.extensions.filters.udp.udp_proxy.session.dynamic_forward_proxy.v3;
+
+import "envoy/extensions/common/dynamic_forward_proxy/v3/dns_cache.proto";
+
+import "google/protobuf/wrappers.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.filters.udp.udp_proxy.session.dynamic_forward_proxy.v3";
+option java_outer_classname = "DynamicForwardProxyProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/filters/udp/udp_proxy/session/dynamic_forward_proxy/v3;dynamic_forward_proxyv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: Filter state dynamic forward proxy]
+
+// Configuration for the filter state based dynamic forward proxy filter. See the
+// :ref:`architecture overview <arch_overview_http_dynamic_forward_proxy>` for
+// more information. Note this filter must be used in conjunction to another filter that
+// sets the 'envoy.upstream.dynamic_host' and the 'envoy.upstream.dynamic_port' filter
+// state keys for the required upstream UDP session.
```

Comment:
Please add a link to the filter that sets these filter state keys.

Reply:
1. Yes this one is a UDP **session** filter only
2. Relevant to previous question too - so Envoy has ``sni_dynamic_forward_proxy`` for network filters, and ``dynamic_forward_proxy`` for HTTP filters. This one adds a similar functionality but for UDP session filters.
3. I'm not sure this doc would be relevant to the UDP variation of the filter, but I indeed added a reference to the [dynamic forward proxy architecture overview](https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/http/http_proxy#arch-overview-http-dynamic-forward-proxy) in the filter docs.
4. Don't have a solid answer here, but possible to distinguish between the kinds of filters and where they can be applied.

> Please add a link to the filter that sets these filter state keys.

There is no filter that sets these filter state keys. The idea is that whoever uses this filter, needs to provide a custom filter that sets the values as they required for their use case - this is specified in the docs I added in this PR. Alternatively, a user could theoretically use Kuat's new filter once it's available #29844.